core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -165,20 +181,31 @@ class RequestSendThread(val controllerId: Int,
     } catch {
       case e: Throwable =>
         error("Controller %d fails to send a request to broker %s".format(controllerId, toBroker.toString()), e)
-        // If there is any socket error (eg, socket timeout), the channel is no longer usable and needs to be recreated.
-        channel.disconnect()
+        // If there is any socket error (eg, socket timeout), the connection is no longer usable and needs to be recreated.
+        selector.close(connectionId)
     }
   }
 
-  private def connectToBroker(broker: Broker, channel: BlockingChannel) {
+  private def connectToBroker() {
     try {
-      channel.connect()
-      info("Controller %d connected to %s for sending state change requests".format(controllerId, broker.toString()))
+      val brokerEndPoint = toBroker.getBrokerEndPoint(config.interBrokerSecurityProtocol)
+      selector.connect(
+        connectionId,
+        new InetSocketAddress(brokerEndPoint.host, brokerEndPoint.port),
+        config.socketSendBufferBytes,
+        config.socketReceiveBufferBytes
+      )
+      val succeeded = SelectorUtils.pollUntil(selector,
+        config.controllerSocketTimeoutMs)(selector.connected().contains(connectionId))(time)
```

Comment:
`config.controllerSocketTimeoutMs` was previously being used for `BlockingChannel.readTimeoutMs`, is it correct to use it here in this way?